off_nitrogens/perturb_angle.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+#=============================================================================================
+# MODULE DOCSTRING
+#=============================================================================================
+
+"""
+perturb_angle.py
+
+Perturb geometry around an improper dihedral angle. The options are:
+ 1. Change valence angle without changing improper
+ 2. Change improper angle without changing valence angles
+
+For use in generating geometries to parameterize valence and improper angles
+
+By: Victoria Lim and Jessica Maat
+
+"""
+
+#=============================================================================================
+# GLOBAL IMPORTS
+#=============================================================================================
+
+import numpy as np
+import math
+import sys
+
+from calc_improper import *
+
+#=============================================================================================
+# PRIVATE SUBROUTINES
```

Comment:
These aren't really private, as in if someone imports your script they get access to these. If you want them to truly be private you can start with an underscore, `_rotation_matrix` instead of `rotation_matrix`. 
I don't think you want to do that here, jut a point on naming.

Reply:
Ah okay. We should rename this then; I think I just lifted the section headings from else where. What would it be called, just "subroutines"?